internal/dao/sqlx/reservation_dao.go

```diff
@@ -10,23 +12,27 @@ import (
 )
 
 const (
-	createReservation         = `INSERT INTO reservations (provider, account_id, status) VALUES ($1, $2, $3) RETURNING *`
-	createAwsDetail           = `INSERT INTO aws_reservation_details (reservation_id, pubkey_id, source_id, instance_type, amount, image_id, poweroff, name) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`
-	updateReservationStatus   = `UPDATE reservations SET status = $2 WHERE id = $1 RETURNING *`
+	createReservation         = `INSERT INTO reservations (provider, account_id, step, steps, status) VALUES ($1, $2, $3, $4, $5) RETURNING *`
+	createAwsDetail           = `INSERT INTO aws_reservation_details (reservation_id, pubkey_id, source_id, image_id, detail) VALUES ($1, $2, $3, $4, $5)`
+	updateReservationStatus   = `UPDATE reservations SET status = $2, step = step + $3 WHERE id = $1 RETURNING *`
 	updateReservationIDForAWS = `UPDATE aws_reservation_details SET aws_reservation_id = $2 WHERE reservation_id = $1 RETURNING *`
-	finishReservationStatus   = `UPDATE reservations SET status = $2, success = $3, finished_at = now() WHERE id = $1 RETURNING *`
+	finishReservationSuccess  = `UPDATE reservations SET success = $2, finished_at = now() WHERE id = $1 RETURNING *`
+	finishReservationError    = `UPDATE reservations SET success = $2, error = $3, finished_at = now() WHERE id = $1 RETURNING *`
 	deleteReservationById     = `DELETE FROM reservations WHERE id = $1`
-	listReservations          = `SELECT * FROM reservations ORDER BY id LIMIT $1 OFFSET $2`
+	getReservationById        = `SELECT * FROM reservations WHERE account_id = $1 AND id = $2 LIMIT 1`
+	listReservations          = `SELECT * FROM reservations WHERE account_id = $1 ORDER BY id LIMIT $2 OFFSET $3`
 	createInstance            = `INSERT INTO reservation_instances (reservation_id, instance_id) VALUES ($1, $2)`
 	listInstanceReservations  = `SELECT * FROM reservation_instances ORDER BY reservation_id LIMIT $1 OFFSET $2`
```

Comment:
I am adding the new fields and details to create statements.

In the update, I do `step = step + X` to do relative update instead of absolute, which cannot be done since the order of jobs is unknown (can be shuffled around later on).

Finish function was split into two variants, one sets status, the other sets error fields.

Both get and list functions are now account-aware as these are the only functions which takes arguments from the user.